### PR TITLE
Don't record built or failed times for repurposed circuits

### PR DESCRIPTION
**Pull request by sporksmith**
When a circuit is cannibalized after being repurposed, it gets an additional `BUILT` or `FAILED` event, which still includes its original `TIME_CREATED`. These events can be long after the original `TIME_CREATED`, which results in confusingly large circuit build times for onion services.

For now ignore these.

https://gitlab.torproject.org/tpo/core/tor/-/issues/40570

### CODE REVIEW

**Comment by sporksmith** [comment]
I tested this in a tornettools CI run. It seems to have the desired effect:

image

I wonder though if maybe we should still process the `CIRC FAILED` events. While the `fail_time` we record here would be wrong, maybe that's better than hiding that such an error happened at all.

OTOOH: as far as I can tell tornettools doesn't actually expose these events? They are recorded as `fail_time` and `circuits_failed_total` here, and tornettools doesn't appear to access those at all.

File: tools/oniontracetools/analysis.py [hunk 1]
```diff
@@ -351,17 +363,26 @@ def __parse_line(self, line, do_complete):
...
+                        cbt = (now_dt - create_dt).total_seconds()
+
+                        if is_built:
+                            self.circuit['build_time'].setdefault(second, [])
+                            self.circuit['build_time'][second].append(cbt)
+                            self.circuit_summary['circuits_built_total'] += 1
+                        else:
+                            self.circuit['fail_time'].setdefault(second, [])
+                            self.circuit['fail_time'][second].append(cbt)
+                            self.circuit_summary['circuits_failed_total'] += 1
```
Comment (gabi-250) [inline tools/oniontracetools/analysis.py]: Could we make `self.circuit_summary['circuits_failed_total']` and `self.circuit_summary['cannibalized_circuits_failed_total']` dictionaries that map failure reasons to the number of times they occurred? Something like:
```python
match = re.search("\sREASON=([^\s]+)", line)
reason = match.group(1)
...
self.circuit_summary['circuits_failed_total'].setdefault(reason, 0)
self.circuit_summary['circuits_failed_total'][reason] += 1
```

Comment (sporksmith) [reply tools/oniontracetools/analysis.py]: Done.

For now I continued to keep separate counts for cannibalized circuits vs non-cannibalized, though merging them would make this code a little nicer.

example:

```
        "circuit_summary": {
          "build_time_max": 3.809,
          "build_time_mean": 0.9578155765765766,
          "build_time_median": 0.949756,
          "build_time_min": 0.124,
          "build_time_std": 0.5330557724211974,
          "cannibalized_circuits_built_total": 68,
          "cannibalized_circuits_failed_total": {
            "DESTROYED": 3
          },
          "circuits_built_total": 111,
          "circuits_failed_total": {
            "TIMEOUT": 7
          },
          "fail_time_max": 1.519864,
          "fail_time_mean": 1.1814874285714285,
          "fail_time_median": 1.089205,
          "fail_time_min": 1.0,
          "fail_time_std": 0.2107927665341058
        }
```

Comment (gabi-250) [reply tools/oniontracetools/analysis.py]: LGTM, thanks!

**Review by robgjansen — APPROVED** [review]

**Pull request merged**